core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -5921,7 +5942,16 @@ private void checkNullOperand(SqlTester tester, String op) {
     tester.checkString("CURRENT_CATALOG", "", "VARCHAR(2000) NOT NULL");
   }
 
-  @Test public void testLocalTimeFunc() {
+  @Tag("slow")
```

Comment:
It might make sense to create `@SlowTest` meta-annotation to avoid typos like `@Tag("solw")`

WDYT?

Reply:
True but I think this will create problems with the dependencies. This actually the reason that I removed the @SlowTest category interface. If for instance we create the annotation `@Slow` in core I think all other modules will depend on it otherwise we cannot use the annotation everywhere. Am I saying something stupid?

Reply:
As you might know, `core/src/test` contains two types of code:
1) Test code. That is `unit tests` or `@Test` methods. This is OK for the code that tests `core`
2) "test framework" code like `CalciteAssert` which is even reused in other modules. Dependency on `-tests.jar` is not that right (currently you can recognize it as dependency on `project(":core", "testClasses")`. For instance, `-tests.jar` do not have their own pom files. I suggest we factor `CalciteAssert`-like test framework code into its own module, and use it as a [test fixture](https://docs.gradle.org/current/userguide/java_testing.html#sec:java_test_fixtures)

Then we could drop `-tests.jar` generation, as test code adds nothing from Maven Central perspective.

Reply:
Yes I thought of this alternative but I would prefer to do it under another JIRA/PR mainly for the following reasons. We have been skipping (complete) slow test execution for quite some time now and each time I get back to this issue there is a new potential regression so I want to get this in ASAP. Let me know if that's OK with you.

Reply:
>  I want to get this in ASAP

@zabetak , are you aware that the current test suite **does** execute the slow tests?
https://travis-ci.org/apache/calcite/jobs/615406555#L1056-L1063

In other words, I look at this change as "refactor-only" rather than "enable running the slow tests"

Reply:
The current suite runs **only a few** of them I think. That explains also the big difference in build times 14m vs 1h 20m.

Reply:
But anyways yes it is mostly refactoring.

Reply:
Ah, I see, there was extra `calcite.test.slow` property.